homeassistant/components/homewizard_energy/config_flow.py

```diff
@@ -0,0 +1,187 @@
+"""Config flow for Homewizard Energy."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohwenergy
+from aiohwenergy.hwenergy import SUPPORTED_DEVICES
+import async_timeout
+from voluptuous import All, Length, Required, Schema
+from voluptuous.util import Lower
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_IP_ADDRESS, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for P1 meter."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Set up the instance."""
+        _LOGGER.debug("config_flow __init__")
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+
+        _LOGGER.debug("config_flow async_step_user")
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=Schema(
+                    {
+                        Required(CONF_IP_ADDRESS): str,
+                    }
+                ),
+                errors=None,
+            )
+
+        entry_info = {
+            CONF_IP_ADDRESS: user_input[CONF_IP_ADDRESS],
+            CONF_PORT: 80,
+        }
+
+        return await self.async_step_check(entry_info)
```

Comment:
No need to ask for confirmation. If the entered IP address works, add it.

Reply:
It now asks for the custom name and then adds it. It won't ask you to confirmation. Maybe the names are wrong.

This can be changed based on what you decide with https://github.com/home-assistant/core/pull/55812#discussion_r762157217

Reply:
Removed confirmation and custom name with https://github.com/home-assistant/core/pull/55812/commits/79f574f2d05a226bddeb6ec3016f667565132e72